lib/acme/client.rb

```diff
@@ -49,6 +49,22 @@ def new_certificate(csr)
     ::Acme::Client::Certificate.new(OpenSSL::X509::Certificate.new(response.body), fetch_chain(response), csr)
   end
 
+  def revoke_certificate(certificate, revocation_key: @private_key)
+    payload = {
+      resource: 'revoke-cert',
+      certificate: Base64.urlsafe_encode64(certificate.to_der)
+    }
+
+    if @private_key.respond_to?(:to_der) && revocation_key.to_der == @private_key.to_der
+      response = connection.post(@operation_endpoints.fetch('revoke-cert'), payload)
+    else
+      tmp_client = ::Acme::Client.new(private_key: revocation_key, endpoint: @endpoint, directory_uri: @directory_uri)
+      response = tmp_client.connection.post(@operation_endpoints.fetch('revoke-cert'), payload)
+    end
+
+    nil
```

Comment:
I would rather return true if it succeed, maybe return `response.success?` from faraday's response object.

This way externally you can use it in conditional.

```ruby
if client.revoke_certificate(..)
  handle_success!
else
  handle_failure!
end
```